mava/environment_loop.py

```diff
@@ -631,14 +614,12 @@ def should_run_loop(eval_interval_condition: Tuple) -> bool:
                     # Check for extra logs
                     if hasattr(self._environment, "get_interval_stats"):
                         results.update(self._environment.get_interval_stats())
-                    self._logger.write(results)
-
                 else:
                     result = self.run_episode()
                     episode_count += 1
                     step_count += result["episode_length"]
                     # Log the given results.
-                    self._logger.write(result)
+                self._logger.write(result)
```

Comment:
Will this not break in the case where `environment_loop_schedule` is not empty? I see that in that case the variable is called results. Maybe we should use the same variable naming?

Reply:
Should we also move this outside the if else?
```
results = self.run_episode()
episode_count += 1
```

Reply:
since I found this command `self._logger.write(result)` will be called in both if statement I thought it is fine to keep it outside the condition

Reply:
Agree I think it works if we move them out of the if statement

Reply:
Yes I agree. The only issue is the naming. See results vs result.

Reply:
Thanks @OmaymaMahjoub :slightly_smiling_face:

Reply:
Oh now it makes sense hh